projects/user_publication/gscholar.py

```diff
@@ -204,8 +203,9 @@ def pub_import(
         cited_pubs = gscholar.get_cites(ch_pub, year_low=year_low, year_high=year_high)
         for cited_pub in cited_pubs:
             pub_model = gscholar.get_pub_model(cited_pub)
-            same_pub = utils.get_publication_with_same_attributes(pub_model, Publication)
-            if same_pub.exists():
-                utils.add_source_to_pub(same_pub.get(), PublicationSource.GOOGLE_SCHOLAR)
+            try:
+                pub_model.year = int(pub_model.year)

```

Comment:
`Publication.year` is a non-null integer field. I would be very surprised if we found a publication without a year, but also, if there _isn't_ a valid year, saving this `Publication` will fail.

Reply:
Yes, it will skip the publication. As for Gscholar, all we are doing is scraping. It sometimes may not have a year to the publication. In my runs, there is one publication for which the whole script will fail, and it's really annoying. So I included the catch here. Will log it and skip the publication.